WebService/ComparisonEngineUI/ComparisonEngineUI/Views/MapPage.xaml.cs

```diff
@@ -6,16 +6,90 @@
 using ComparisonEngineUI.ViewModels;
 using Xamarin.Forms;
 using Xamarin.Forms.Xaml;
+using System.Diagnostics;
+using Xamarin.Forms.Maps;
+using ComparisonEngineUI.Models;
 
 namespace ComparisonEngineUI.Views
 {
     [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class MapPage : ContentPage
     {
+        Map map;
+        Xamarin.Essentials.Location currentPosition;
         public MapPage()
         {
             InitializeComponent();
-            this.BindingContext = new MapViewModel();
+            BindingContext = new MapViewModel();
+            
+            map = new Map
+            {
+
+                HeightRequest = 100,
+                WidthRequest = 960,
+                VerticalOptions = LayoutOptions.FillAndExpand
+            };
+
+            GetCurrentPosition();
+
+            if (currentPosition != null)
+            {
+                map.MoveToRegion(MapSpan.FromCenterAndRadius(new Position(currentPosition.Latitude, currentPosition.Longitude), Distance.FromMiles(3)));
+            }
+            else
+            {
+                map.MoveToRegion(MapSpan.FromCenterAndRadius( new Position(54.68916, 25.2798), Distance.FromMiles(3)));
+            }
+
+            map.IsShowingUser = true;
+
+            var stack = new StackLayout { Spacing = 0 };
+            stack.Children.Add(map);
+            Content = stack;
+        }
+
+        public void SetDrinkPins(List<Bar> items)
+        {
+            foreach (Bar item in items)
+            {
+                SetBarPin(item);
+            }
+        }
```

Comment:
not sure if this should be a List<Bar>, but can't think of anything better at this point